siliconcompiler/fpgas/lattice_ice40.py

```diff
@@ -0,0 +1,33 @@
+import siliconcompiler
+
+
+####################################################
+# Setup for ICE40 Family FPGAs
+####################################################
+def setup(chip):
+    '''
+    Lattice ICE40 FPGAs are a family of small parts
+    made by Lattice Semiconductor.  A fully open source
+    RTL to bitstream flow for ICE40 is available using
+    yosys + nextpnr
+    '''
+
+    family = 'ice40'
+    vendor = 'lattice'
+
+    lut_size = '4'
+
+    fpga = siliconcompiler.FPGA(chip, family)
+
+    fpga.set('fpga', 'vendor', vendor)
+    fpga.set('fpga', family, 'lutsize', lut_size)
+
+    chip.set('tool', 'yosys', 'task', 'syn', 'var', 'lut_size', f'{lut_size}')
```

Comment:
if we set the lutsize for the family, do we need to set it for the tool? The tool driver should use the fpga/family/lutsize parameter